### PR TITLE
ufrontend

### PR DESCRIPTION
**Pull request by cbourgois**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | yes
| Tickets          | Fix #... 
| License          | BSD 3-Clause

## Before merging

This pull request adds the micro frontend framework and navbar fragment to the mono repo. It does not integrate them to any application. Since this work involves some changes to manager-core and manager-config but also to the dev server. Some QC is needed to ensure that no side effect or bug is present on current applications.

## Description

// To Complete

* Add `@ovh-ux/ufrontend`: micro frontend framework to allow dynamic loading of fragments into applications (with communication layer)
* move `bootstrapApplication` from `@ovh-ux/manager-core` to `@ovh-ux/manager-config`
* add `@ovh-ux/manager-registry` package with registry CLI
  * with `dev` sub command to allow to serve local dev fragments
  * with `static` sub command to manage a static registry:
    * `static generate-manifests` to generate manifests files from a folder containing fragments
    * `static serve` command to serve a static registry from folder
    * `static add` command to add a fragment dist to a registry
* update `@ovh-ux/manager-dev-server-config`
  * add registry proxy
* update `@ovh-ux/manager-dev-server`, update `@ovh-ux/manager-webpack-dev-server`
  * add `localRegistry` option to use a local registry
  * add `registryUrl` option to configure a registry url
* add `@ovh-ux/manager-fragment-navbar`: rewrite of current navbar to a fragment, as defined by our micro frontend framework
* update root `build` script to allow to build packages for a specific workspace.

## Related PRs

* [x] https://github.com/ovh/manager/pull/3281
* [x] https://github.com/ovh/manager/pull/3257
* [x] https://github.com/ovh/manager/pull/3237
* [x] https://github.com/ovh/manager/pull/3245
* [x] https://github.com/ovh/manager/pull/3246
* [x] https://github.com/ovh/manager/pull/3302 (we need to retrieve the locale from the config to give it to fragments)
* [x] https://github.com/ovh/manager/pull/3193 (move account-sidebar out of navbar) 
* [x] https://github.com/ovh/manager/pull/3725 (add `workspace` option in `build` script)

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build#14236.0 ✘  
* 
  * Build ✘

**Comment by frenautvh** [comment]
rebase on develop

**Review by antleblanc — CHANGES_REQUESTED** [review]
@cbourgois In order to be able to merge this Pull Request, could you please resolve existing conflict?

Thank you!

**Comment by cbourgois** [comment]
Rebased on `develop`

File: packages/manager/fragments/navbar/src/translations/navbar_fr_FR.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "navbar_universe_sunrise": "Sunrise",
+  "navbar_universe_hub": "Tableau de bord",
+  "navbar_universe_partners": "Partenaires",
+  "navbar_universe_public-cloud": "Public Cloud",
+  "navbar_universe_server": "Server",
```
Comment (antleblanc) [inline packages/manager/fragments/navbar/src/translations/navbar_fr_FR.json:6]:
```suggestion
  "navbar_universe_server": "Bare Metal  Cloud",
```

File: packages/manager/fragments/navbar/src/translations/navbar_fr_FR.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  "navbar_universe_telecom": "Télécom",
+  "navbar_universe_web": "Web",
```
Comment (antleblanc) [inline packages/manager/fragments/navbar/src/translations/navbar_fr_FR.json:8]:
```suggestion
  "navbar_universe_web": "Web Cloud",
```

File: packages/manager/fragments/navbar/src/translations/navbar_fr_FR.json (hunk 1 above)
Comment (antleblanc) [reply packages/manager/fragments/navbar/src/translations/navbar_fr_FR.json:6]: **Note:** Missing `Hosted Private Cloud`

File: packages/manager/modules/ufrontend/.eslintrc.json [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{
```
Comment (antleblanc) [inline packages/manager/modules/ufrontend/.eslintrc.json:1]: **Note:** This file can be removed.

File: packages/manager/modules/ufrontend/src/utils/deferred.class.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+ * foo.isPending(); // true
+ * foo.resolve('bar');
+ * foo.isPending(); // false
+ * foo.promise.then(x => ...) // x = 'bar'
+ *
+ * // with initial value
+ * const hello = Deferred('value');
+ * foo.isPending(); // true
+ * foo.resolve();
+ * foo.promise.then(x => ...) // x = 'value'
```
Comment (antleblanc) [inline packages/manager/modules/ufrontend/src/utils/deferred.class.js:16]:
```suggestion
 * hello.isPending(); // true
 * hello.resolve();
 * hello.promise.then(x => ...) // x = 'value'
```

File: packages/manager/tools/registry/README.md [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+
+```sh
+$ manager-registry static add ./path/to/static/registry ./path/to/fragment/dist
+Fragment "fragment@1.0.O" already exists in ./path/to/static/registry
+```
+
+
+## Related
+
+* [manager-registry](https://github.com/ovh-ux/manager/tree/master/packages/manager/tools/registry) - OVHcloud manager shared dev server configuration
```
Comment (antleblanc) [inline packages/manager/tools/registry/README.md:202]: Wrong link I guess.

Isn't supposed to be:
```suggestion
* [dev-server-config](https://github.com/ovh/manager/tree/master/packages/manager/tools/dev-server-config) - OVHcloud manager shared dev server configuration
```

**Review by antleblanc — APPROVED** [review]

File: packages/manager/tools/registry/README.md (hunk 5 above)
Comment (cbourgois) [reply packages/manager/tools/registry/README.md]: Fixed.

File: packages/manager/modules/ufrontend/src/utils/deferred.class.js (hunk 4 above)
Comment (cbourgois) [reply packages/manager/modules/ufrontend/src/utils/deferred.class.js]: Fixed.

File: packages/manager/modules/ufrontend/.eslintrc.json (hunk 3 above)
Comment (cbourgois) [reply packages/manager/modules/ufrontend/.eslintrc.json]: Removed.

File: packages/manager/fragments/navbar/src/translations/navbar_fr_FR.json (hunk 2 above)
Comment (cbourgois) [reply packages/manager/fragments/navbar/src/translations/navbar_fr_FR.json]: Fixed + `Bare Metal Cloud`

File: packages/manager/fragments/navbar/src/translations/navbar_fr_FR.json (hunk 1 above)
Comment (cbourgois) [reply packages/manager/fragments/navbar/src/translations/navbar_fr_FR.json]: Fixed (both)

**Review by antleblanc — APPROVED** [review]

**Pull request merged**